### DISCUSSION TITLE
`notifications` does not work in NextJS 13 `app` router

### DISCUSSION THREAD

**Discussion by shahryarjb** [mantinedev/mantine#4857] [Q&A]
Hello, I can not be able to show a notification inside my NextJS website, but `onClose` is worked if `autoClose` is enabled. I think I have problem with the graphic of the notification.

### Step one
in my `RootLayout` I add a wrapper of `Notifications` component
```javascript
  return (
    <html
      lang={`${locale}`}
      className={`${persianFont.variable} ${poppinsFont.variable}`}
      dir={`${locale === 'fa' ? 'rtl' : 'ltr'}`}
    >
      <body>
        <NextAuthProvider>
          <MantineProvider theme={theme}>
            <NotificationsWrapper />
            <section id="login" className="lg:min-h-screen flex flex-col">
              <Nav dict={dict} />
              {children}
            </section>
          </MantineProvider>
        </NextAuthProvider>
      </body>
    </html>
  );
```

### Step two
And the wrapper component is:

```javascript
'use client';

import { Notifications } from '@mantine/notifications';
import '@mantine/notifications/styles.css';

const NotificationsWrapper = () => {
  return <Notifications limit={5} zIndex={1000} />;
};

export default NotificationsWrapper;
```

## Finally
```javascript
import { showNotification } from '@mantine/notifications';

const LoginGetOTPFormComponent = ({ dict }: { dict: any }) => {
  const emailRef: RefObject<HTMLInputElement> = useRef(null);

  const otpHandler = useMutation(getOTPRequest, {
    onSuccess: () => {
      // navigating
    },
    onError: ({ response }) => {
      // Error handler
      showNotification({
        id: 'hello-there25',
        withCloseButton: true,
        autoClose: 5000,
        title: "You've been compromised",
        message: 'Leave the building immediately',
        color: 'red',
        className: 'my-notification-class',
        style: { backgroundColor: 'red' },
        loading: false,
      });
    },
  });
```

I tried with a normal button of Nextjs and does not work like top code

Versions: 
- `"@mantine/notifications": "^7.0.0"`
- `"next": "latest"`

It should be noted the other component like `dropzone` and `carousel` work in my app and has no error

Thank you

**Comment by rtivital** [comment]
Please make sure that:
- Your `onError` function is actually called
- Create a button that calls showNotification function when clicked, if notification is shown then the issue is somewhere else

**Reply by shahryarjb** [reply]
@rtivital Thank you, yes it called every time i send request to server

and I create an button or a normal div like this

```javascript
import { showNotification } from '@mantine/notifications';

    <div
      onClick={() =>
        showNotification({
          title: 'Hey this is',
          message: 'Wo this is what it is',
        })
      }
    >
      Show notification
    </div>
```

Still did not work, it should be noted

```javascript
  const otpHandler = useMutation(getOTPRequest, {
    onSuccess: () => {
      // navigating
    },
    onError: ({ response }) => {
      const createdID = generateRandomString()
      console.log("the id", createdID)
      // Error handler
      showNotification({
        id: createdID,
        withCloseButton: true,
        onClose: () => console.log('unmounted =>>>', createdID),
        autoClose: 5000,
        title: `You've been compromised ${createdID}`,
        message: `Leave the building immediately ${createdID}`,
        color: 'red',
        className: 'my-notification-class',
        style: { backgroundColor: 'red' },
        loading: true,
      });
    },
  });
```

After the `autoClose: 5000` time, I can see the `console.log('unmounted =>>>', createdID)` in my console

I can share the project if it is needed

Thank you in advanced

**Reply by shahryarjb** [reply]
Hey again, I disabled all my blocks and I see the Notification, it seems has no style. I do not know why it has no style because I imported the css file of documents. my web app is created by TailwindCSS

![Screenshot 2023-09-25 at 00 44 15](https://github.com/mantinedev/mantine/assets/8413604/71eeedf9-5d7b-411f-b0f6-abdf4c515520)

Do you have any suggestion?
@rtivital

**Reply by rtivital** [reply]
Have you imported `@mantine/notifications/styles.css`?

**Reply by shahryarjb** [reply]
> Have you imported `@mantine/notifications/styles.css`?

Yes as I put the code on top. but my problem is fixed, I just deleted `.next` and it works.
Thank you
